### PR TITLE
Restore check for non-existent path

### PR DESCRIPTION
**Pull request by kylog**

### CODE REVIEW

File: lib/puppet/node/environment.rb [hunk 1]
```diff
@@ -318,13 +318,17 @@ def warn_about_mistaken_path(path, name)
   def modules_by_path
     modules_by_path = {}
     modulepath.each do |path|
-      Dir.chdir(path) do
-        module_names = Dir.entries(path).select do |name|
-          Puppet::Module.is_module_directory?(name, path)
-        end
-        modules_by_path[path] = module_names.sort.map do |name|
-          Puppet::Module.new(name, File.join(path, name), self)
+      if File.exists?(path)
```
Comment (joshcooper) [inline lib/puppet/node/environment.rb]: `Puppet::FileSystem.exists?(path)`?

**Comment by branan** [comment]
There's a race condition here, but that's also in the current PE codebase. Good to hit parity, and we can sort out cleaning this mess up another day.

**Comment by kylog** [comment]
@joshcooper good call; fix pushed.

@branan yeah, I created https://tickets.puppetlabs.com/browse/PUP-4713. There's a lot not to like about that code, and we may reach the point where a PMT rewrite is prefereable to servicing its tech debt.

**Comment by MikaelSmith** [comment]
AppVeyor failure is rdoc.

**Pull request merged**